src/app/src/main/java/com/example/bookmark/BorrowedActivity.java

```diff
@@ -78,22 +79,48 @@ protected boolean getBookStatusVisibility() {
      */
     @Override
     protected void getBooks() {
-        StorageServiceProvider.getStorageService().retrieveBooksByRequester(
-                user,
-                books -> {
-                    visibleBooks.clear();
-                    relevantBooks.clear();
-                    for (Book book : books) {
-                        if (book.getStatus() == Book.Status.BORROWED) {
-                            relevantBooks.add(book);
+        if (user == null) {
+            String username = UserUtil.getLoggedInUser(this);
+            StorageServiceProvider.getStorageService().retrieveUserByUsername(username,
+                    user1 -> {
+                        StorageServiceProvider.getStorageService().retrieveBooksByRequester(
+                                user1,
```

Comment:
May want to set `this.user` to `user1` after it is retrieved by `retrieveUserByUsername` and then use `this.user` here instead of `user1`. `user` would still probably get set ListingBooksActivity  but under strange timing conditions it may cause more than just two network calls.

Reply:
Good thinking